htdocs/xoops_lib/smarty/xoops_plugins/function.translate.php

```diff
@@ -18,8 +18,16 @@
 
 function smarty_function_translate($params, &$smarty)
 {
```

Comment:
Probably better to use a separate array for the value pairs for formatting instead of restricting the key space? Arrays are simple in Smarty 3. Maybe something that looks more like this:
```
{assign myvalues ['name' => 'Smarty',  'gender' = 'template engine']}
{translate key="GENDER" dirname="codex" values=$myvalues}
```